tools/generate_spark_test_code.py

```diff
@@ -55,9 +58,13 @@ def main() -> int:
         output_directory = feature_test / "generated"
         output_directory.mkdir(exist_ok=True)
         parser = Parser()
-        parser.parse(feature_test / "test.rflx")
+        parser.parse(*(feature_test.glob("*.rflx")))
```

Comment:
This change doesn't make sense without changing `feature_test.py`, as only a `test.rflx` is expected there. Changing `feature_test.py`, so that multiple entry points are supported, could get quite complex. I don't think we should do that now.

Reply:
I agree that this change alone is insufficient. But without it I can't write an integration test like the one in `session_integration_multiple`. I guess you would want to see that as a unit test, but the effort is really high compared to the approach I took here.
I think we should change the test framework to accommodate for tests like this.

Reply:
No, I would not recommend writing a complex unit test. That would contradict our definition: "A unit test is simple, fast and generic".

Adding real multiple session support to `feature_test.py` would not be a simple change. I also don't think the test framework needs to be changed for the desired test. By changing the name of one spec to `test.rflx` and importing the other, the compilability test could be run:

```diff
diff --git a/tests/integration/session_integration_multiple/a.rfi b/tests/integration/session_integration_multiple/a.rfi
deleted file mode 100644
index 18e95fa47..000000000
--- a/tests/integration/session_integration_multiple/a.rfi
+++ /dev/null
@@ -1,6 +0,0 @@
-Session:
-  Session:
-    Buffer_Size:
-      Local:
-        Start:
-          M: 1024
diff --git a/tests/integration/session_integration_multiple/b.rfi b/tests/integration/session_integration_multiple/b.rfi
index e412d43d2..2ac266c6b 100644
--- a/tests/integration/session_integration_multiple/b.rfi
+++ b/tests/integration/session_integration_multiple/b.rfi
@@ -1,6 +1,5 @@
 Session:
   Session:
     Buffer_Size:
-      Local:
-        Start:
-          M: 2048
+      Global:
+        M: 2048
diff --git a/tests/integration/session_integration_multiple/b.rflx b/tests/integration/session_integration_multiple/b.rflx
index d943fc8be..cc2da7b7b 100644
--- a/tests/integration/session_integration_multiple/b.rflx
+++ b/tests/integration/session_integration_multiple/b.rflx
@@ -7,9 +7,9 @@ package B is
       Initial => Start,
       Final => Terminated
    is
+      M : Msg::Message;
    begin
       state Start is
-         M : Msg::Message;
       begin
          Channel'Read (M);
       transition
diff --git a/tests/integration/session_integration_multiple/test.rfi b/tests/integration/session_integration_multiple/test.rfi
new file mode 100644
index 000000000..d37c5b555
--- /dev/null
+++ b/tests/integration/session_integration_multiple/test.rfi
@@ -0,0 +1,5 @@
+Session:
+  Session:
+    Buffer_Size:
+      Global:
+        M: 1024
diff --git a/tests/integration/session_integration_multiple/a.rflx b/tests/integration/session_integration_multiple/test.rflx
similarity index 84%
rename from tests/integration/session_integration_multiple/a.rflx
rename to tests/integration/session_integration_multiple/test.rflx
index 01bce984a..0193a9524 100644
--- a/tests/integration/session_integration_multiple/a.rflx
+++ b/tests/integration/session_integration_multiple/test.rflx
@@ -1,5 +1,7 @@
 with Msg;
-package A is
+with B;
+
+package Test is
 
    generic
       Channel : Channel with Readable;
@@ -7,9 +9,9 @@ package A is
       Initial => Start,
       Final => Terminated
    is
+      M : Msg::Message;
    begin
       state Start is
-         M : Msg::Message;
       begin
          Channel'Read (M);
       transition
@@ -19,4 +21,4 @@ package A is
       state Terminated is null state;
    end Session;
 
-end A;
+end Test;
```

Reply:
Fair enough, I guess in most cases we can get away this "fake" entry point. I will do what you suggest. A question, why do you transform the variable into a global variable?

Reply:
Because local declarations are not supported in IO states (cf. #858).

Reply:
fixed in 855f0000e25bb718f8dee56e5e15d38e1128aa81